### PR TITLE
GSoC: Week 8 Blogpost

### PR DESCRIPTION
**Pull request by JoaoDell**
Hello everyone, here is my week 8 blogpost:
image

### CODE REVIEW

File: docs/source/posts/2023/2023-07-24-week-8-joaodellagli.rst [hunk 1]
```diff
@@ -0,0 +1,119 @@
...
+   :category: gsoc
+
+
+Hello everyone, it's time for another weekly blogpost! Today, I am going to tell you all about how is the KDE API development going, and 
+to show you the potential this holds for the future!
+
+Last Week's Effort
+------------------
+Last week I told you how I managed to render some KDE renders to the screen, both in 2D and 3D, as you may check by my last blogpost. 
+My new task was, as I had this example working, to start the API development. I;n a meeting with Bruno, one of my mentors, we debated
```
Comment (tvcastillod) [inline docs/source/posts/2023/2023-07-24-week-8-joaodellagli.rst:16]: **_In_** a meeting

File: docs/source/posts/2023/2023-07-24-week-8-joaodellagli.rst [hunk 2]
```diff
@@ -0,0 +1,119 @@
...
+------------------
+Last week I told you how I managed to render some KDE renders to the screen, both in 2D and 3D, as you may check by my last blogpost. 
+My new task was, as I had this example working, to start the API development. I;n a meeting with Bruno, one of my mentors, we debated 
+on how could this work, reaching two options:
+
+1. Implement the KDE in a single, simple actor.
+2. Implement a KDE rendering manager, as a class.
+
+The first one would have the advantage of being simple and pretty straightforward, as a user would only need to call the actor and have 
+it working on them hands, having the tradeoff of leaving some important steps for a clean API hidden and static. These steps I mention
```
Comment (tvcastillod) [inline docs/source/posts/2023/2023-07-24-week-8-joaodellagli.rst:23]: working on **_their_** hands

File: docs/source/posts/2023/2023-07-24-week-8-joaodellagli.rst [hunk 3]
```diff
@@ -0,0 +1,119 @@
...
+1. Implement the KDE in a single, simple actor.
+2. Implement a KDE rendering manager, as a class.
+
+The first one would have the advantage of being simple and pretty straightforward, as a user would only need to call the actor and have 
+it working on them hands, having the tradeoff of leaving some important steps for a clean API hidden and static. These steps I mention 
+are related to how this rendering works, as I have previously :doc:`showed you <2023-07-03-week-5-joaodellagli>`, it relies on post-processing effects, 
+which need an offscreen rendering, that for example are done by the *callback functions*. 
+
+In short, these functions are instructions the user gives to the interactor to run inside the interaction loop. Inside FURY there are tree
+types of callbacks passed to the windoe interactor:
```
Comment (tvcastillod) [inline docs/source/posts/2023/2023-07-24-week-8-joaodellagli.rst:28]: to the **_window_** interactor

File: docs/source/posts/2023/2023-07-24-week-8-joaodellagli.rst [hunk 4]
```diff
@@ -0,0 +1,119 @@
...
+3. **Interactor Callbacks**: Added to the window interactor, they are a set of instructions called whenever an specific interaction, for example a mouse left-click, is triggered. 
+
+In this API, I will be using the *Interactor Callback*, set by the ``window.add_iren_callback()`` function, that will be called whenever a *Render* 
+interaction is detected, and needs to be first passed to the onscreen manager. 
+
+These details are more complicated, and would need, for example, for the user to pass the onscreen manager to the ``actor.kde()`` function. 
+Also, in the case of a kde actor not being used anymore and being declared, the callback then passed would still exist inside the manager and
+be called even when the kde actor is not on screen anymore, which is not ideal;
+
+Knowing these problems, we thought of second option, that would have the advantage of not leaving those details and steps behind. It has
```
Comment (tvcastillod) [inline docs/source/posts/2023/2023-07-24-week-8-joaodellagli.rst:41]: we thought of **_a_** second option

File: docs/source/posts/2023/2023-07-24-week-8-joaodellagli.rst [hunk 5]
```diff
@@ -0,0 +1,119 @@
...
+These details are more complicated, and would need, for example, for the user to pass the onscreen manager to the ``actor.kde()`` function. 
+Also, in the case of a kde actor not being used anymore and being declared, the callback then passed would still exist inside the manager and
+be called even when the kde actor is not on screen anymore, which is not ideal;
+
+Knowing these problems, we thought of second option, that would have the advantage of not leaving those details and steps behind. It has 
+the tradeoff of maybe complicating things as it would need to be called after calling the effects manager, but as I will show you below, 
+it is not that complicated *at all*. 
+
+I also reviewed my fellow GSoC contributors PR's as well, PR `#810 <https://github.com/fury-gl/fury/pull/810>`_ and 
+`#803 <https://github.com/fury-gl/fury/pull/803>`_. Bruno told me to take a look as well on `Conventional Commits <https://www.conventionalcommits.org>`_ , a way to standardtize
```
Comment (tvcastillod) [inline docs/source/posts/2023/2023-07-24-week-8-joaodellagli.rst:46]: a way to _**standardize**_

File: docs/source/posts/2023/2023-07-24-week-8-joaodellagli.rst [hunk 6]
```diff
@@ -0,0 +1,119 @@
...
+point:
+
+.. image:: https://raw.githubusercontent.com/JoaoDell/gsoc_assets/main/images/3d_kde_gif.gif
+   :align: center
+   :alt: 3D KDE render
+
+After having that working, I experimented beyond. See, as I previously said, we are dealing here with *post-processing effects*, with KDE
+being only one of the many existing ones, as this `Wikipedia Page <https://en.wikipedia.org/wiki/Video_post-processing>`_ on post processing shows. 
+Knowing that, I tried one of the first filters I learned, the Laplacian one. This filter is, as its name hints, applying the 
+`Discrete Laplace Operator <https://en.wikipedia.org/wiki/Discrete_Laplace_operator>`_ in an image. This filter shows suddend changes of value, a
```
Comment (tvcastillod) [inline docs/source/posts/2023/2023-07-24-week-8-joaodellagli.rst:87]: shows **_sudden_** changes

File: docs/source/posts/2023/2023-07-24-week-8-joaodellagli.rst [hunk 7]
```diff
@@ -0,0 +1,119 @@
...
+   :alt: Laplacian filter applied to a cube object.
+
+Something I found important to leave as an option was filter compositing. What if an user wanted to, for example, apply one laplacian filter
+after another? Well, the example below shows that is possible as well:
+
+.. image:: https://raw.githubusercontent.com/JoaoDell/gsoc_assets/main/images/laplacian2.gif
+   :align: center
+   :alt: Double laplacian application on the box actor.
+
+It still need some tweaks and suffers from some bugs, but it works! Those represent important progress as it shows the versatility this
```
Comment (tvcastillod) [inline docs/source/posts/2023/2023-07-24-week-8-joaodellagli.rst:102]: It still _**needs**_ some

**Review by tvcastillod — COMMENTED** [review]
Hi @JoaoDell 
Looks like you made great progress last week, that's great! The blogpost is generating correctly and looks great overall, I just left some comments on spelling issues.

**Review by ganimtron-10 — COMMENTED** [review]
Hey @JoaoDell ,
Everything LGTM.
Nice Blogpost.

File: docs/source/posts/2023/2023-07-24-week-8-joaodellagli.rst [hunk 8]
```diff
@@ -0,0 +1,119 @@
...
+commits by prefixes, so I did that as well.
+
+So how did it go?
+-----------------
+
+Well, the implemented manager class is named ``EffectManager()`` and to initialize it you only need to pass the onscreen manager:
+
+.. code-block:: python
+
+   effects = EffectManager(onscreen_manager)
```
Comment (skoudoro) [inline docs/source/posts/2023/2023-07-24-week-8-joaodellagli.rst:56]: What is `onscreen_manager` ? you do not talk about it before. Maybe just add one line of code to show how you initialize this object.

**Review by skoudoro — COMMENTED** [review]
Hi @JoaoDell,

I just have one small comment below, otherwise, it looks good and ready to be merged as soon as you address my comment.

**Comment by JoaoDell** [comment]
@skoudoro just fixed that, see if it is better now. Thanks!

**Review by skoudoro — APPROVED** [review]
Thanks for the update @JoaoDell, merging.

**Pull request merged**